components/avatar/avatar.vue

```diff
@@ -103,36 +94,108 @@ export default {
       type: Object,
       default: () => ({}),
     },
+
+    /**
+     * Initials to be shown in the avatar. Used as fallback if image fails to load.
+     */
+    initials: {
+      type: String,
+      default: '',
+    },
   },
 
   data () {
     return {
       // initials, image or icon
       kind: 'initials',
       AVATAR_SIZE_MODIFIERS,
-      AVATAR_COLOR_MODIFIERS,
       AVATAR_KIND_MODIFIERS,
       AVATAR_PRESENCE_SIZE_MODIFIERS,
+      imageLoadedSuccessfully: null,
+      formattedInitials: '',
     };
   },
 
+  computed: {
+    shouldShowContent () {
+      return this.kind !== 'initials' && this.imageLoadedSuccessfully !== false;
+    },
+
+    showInitials () {
+      return this.kind === 'initials' || this.initials;
+    },
+
+    initialKindStyle () {
+      const randomGradientColorStops = this.randomizeGradientColorStops();
+      const styles = {
+        '--avatar-gradient-angle': `${this.randomizeGradientAngle()}deg`,
+        '--avatar-gradient-stop-1': `var(--${randomGradientColorStops[0]})`,
+        '--avatar-gradient-stop-2': `var(--${randomGradientColorStops[1]})`,
+        '--avatar-gradient-stop-3': `var(--${randomGradientColorStops[2]})`,
+      };
+
+      return (this.kind === 'initials' || this.kind === 'icon') ? styles : null;
+    },
+  },
+
   mounted () {
     this.init();
+    this.validateImageAttrsPresence();
   },
 
   updated () {
     this.init();
   },
 
   methods: {
-    init () {
-      const firstChild = this.$el.firstChild;
-      if (firstChild) {
-        this.setKind(firstChild);
-        if (this.kind === 'image') {
-          firstChild.classList.add('d-avatar__image');
+    async init () {
+      const firstChild = this.$slots.default[0]?.elm || this.$slots.default[0];
```

Comment:
It's for getting the html in the slot since the default slot isn't rendered until the kind is defined. Since it's an internal prop from the vnode, will try to refactor just using the exposed `$el` prop

Reply:
alright cool, I just had never seen it before was curious what it was.

Reply:
The challenge found here was the logic for showing initials/fallback to initials when the image doesn't load and having all the variants supported by the default slot. If it was supported by a prop and thus reactive, things might be easier.